src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1077 @@
+	installing: localize('welcomePage.installingExtensionPack', "Installing additional support for {0}..."),
+	extensionNotFound: localize('welcomePage.extensionPackNotFound', "Support for {0} with id {1} could not be found."),
+};
+
+
+const welcomeInputTypeId = 'workbench.editors.welcomePageInput';
+class WelcomePage extends Disposable {
+	readonly editorInput: WalkThroughInput;
+	constructor(
+		@IEditorService private readonly editorService: IEditorService,
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IWorkspacesService private readonly workspacesService: IWorkspacesService,
+		@IWorkspaceContextService private readonly contextService: IWorkspaceContextService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@ILabelService private readonly labelService: ILabelService,
+		@INotificationService private readonly notificationService: INotificationService,
+		@IWorkbenchExtensionEnablementService private readonly extensionEnablementService: IWorkbenchExtensionEnablementService,
+		@IExtensionGalleryService private readonly extensionGalleryService: IExtensionGalleryService,
+		@IExtensionManagementService private readonly extensionManagementService: IExtensionManagementService,
+		@IExtensionTipsService private readonly tipsService: IExtensionTipsService,
+		@IExtensionsWorkbenchService private readonly extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@ILifecycleService lifecycleService: ILifecycleService,
+		@ITelemetryService private readonly telemetryService: ITelemetryService,
+		@IHostService private readonly hostService: IHostService,
+		@IFileService fileService: IFileService,
+		@IProductService private readonly productService: IProductService,
+	) {
+		super();
+		this._register(lifecycleService.onShutdown(() => this.dispose()));
+		const recentlyOpened = this.workspacesService.getRecentlyOpened();
+		const installedExtensions = this.instantiationService.invokeFunction(getInstalledExtensions);
+		const resource = URI.parse(require.toUrl('./az_data_welcome_page'))
+			.with({
+				scheme: Schemas.walkThrough,
+				query: JSON.stringify({ moduleId: 'sql/workbench/contrib/welcome/page/browser/az_data_welcome_page' })
+			});
+		this.editorInput = this.instantiationService.createInstance(WalkThroughInput, {
+			typeId: welcomeInputTypeId,
+			name: localize('welcome.title', "Welcome"),
+			resource,
+			telemetryFrom,
+			onReady: (container: HTMLElement) => this.onReady(container, recentlyOpened, installedExtensions, fileService)
+		});
+	}
+	public openEditor() {
+		return this.editorService.openEditor(this.editorInput, { pinned: false });
+	}
+	private onReady(container: HTMLElement, recentlyOpened: Promise<IRecentlyOpened>, installedExtensions: Promise<IExtensionStatus[]>, fileService: IFileService): void {
```

Comment:
There is a lot of mixing and matching of async-await and then here. Not an issue but it could look cleaner if you just picked async await.